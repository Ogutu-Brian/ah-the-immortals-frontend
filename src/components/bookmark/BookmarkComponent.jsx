import React, {Component} from 'react';
import PropTypes from 'prop-types';

class BookmarkComponent extends Component {
  static  propTypes = {
    article: PropTypes.object.isRequired,
    handleClick: PropTypes.func.isRequired
  };
  render() {
    const { article, handleClick } = this.props;

    return (
      <div className={article.bookmarked?'bookmarked-icon':'bookmark'}>
        <i className="fas fa-bookmark" onClick={()=>handleClick(article.slug)} />
      </div>
    );
  }
}

export default BookmarkComponent;
