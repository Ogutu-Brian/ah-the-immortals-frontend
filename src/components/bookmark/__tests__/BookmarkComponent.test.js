import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import BookmarkComponent from '../BookmarkComponent';



describe('Test BookmarkComponent', () => {
  const props = {
    article: {
      bookmarked: true
    },
    handleClick : jest.fn
  };

  const wrapper = mount(
    <BookmarkComponent {...props} />
  );

  it('test view profile container snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
