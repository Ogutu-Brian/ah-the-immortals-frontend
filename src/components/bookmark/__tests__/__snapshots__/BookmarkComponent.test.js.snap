// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BookmarkComponent test view profile container snapshot 1`] = `
<BookmarkComponent
  article={
    Object {
      "bookmarked": true,
    }
  }
  handleClick={[Function]}
>
  <div
    className="bookmarked-icon"
  >
    <i
      className="fas fa-bookmark"
      onClick={[Function]}
    />
  </div>
</BookmarkComponent>
`;
